sbc_compassion/models/sponsorship_correspondence.py

```diff
@@ -172,38 +171,50 @@ def get_communication_types(self):
             ('Supporter Letter', _('Supporter Letter')),
         ]
 
+    @api.multi
     @api.depends('sponsorship_id')
     def _set_name(self):
-        if self.sponsorship_id and self.communication_type_ids:
-            self.name = self.communication_type_ids[0].name + ' (' + \
-                self.sponsorship_id.partner_codega + " - " + \
-                self.child_id.code + ')'
-        else:
-            self.name = _('New correspondence')
+        for corr in self:
+            if corr.sponsorship_id and corr.communication_type_ids:
+                corr.name = corr.communication_type_ids[0].name + ' (' + \
+                            corr.sponsorship_id.partner_codega + " - " + \
+                            corr.child_id.code + ')'
+            else:
+                corr.name = _('New correspondence')
 
-    @api.depends('sponsorship_id', 'direction')
+    @api.depends('sponsorship_id', 'direction', 'original_language_id',
+                 'destination_language_id')
     def _set_languages(self):
-        if self.direction == 'Supporter To Beneficiary':
-            if self.correspondant_id.spoken_langs_ids:
-                self.original_language_id = self.correspondant_id\
-                    .spoken_langs_ids[0]
-            if self.child_id.project_id.country_id.spoken_langs_ids:
-                self.destination_language_id = self.child_id.project_id\
-                    .country_id.spoken_langs_ids[0]
-        if self.direction == 'Beneficiary To Supporter':
-            if self.child_id.project_id.country_id.spoken_langs_ids:
-                self.original_language_id = self.child_id.project_id\
-                    .country_id.spoken_langs_ids[0]
-            if self.correspondant_id.spoken_langs_ids:
-                self.destination_language_id = self.correspondant_id\
-                    .spoken_langs_ids[0]
+        for corr in self:
+            if corr.direction == 'Supporter To Beneficiary':
+                if corr.child_id.project_id.country_id.spoken_langs_ids:
+                    if corr.original_language_id in corr.child_id.project_id.\
+                       country_id.spoken_langs_ids:
+                        corr.destination_language_id = corr.\
+                                                       original_language_id
+                    else:
+                        corr.destination_language_id = corr.child_id\
+                                                           .project_id\
+                                                           .country_id\
+                                                           .spoken_langs_ids[0]
 
-    @api.depends('sponsorship_id')
+            if corr.direction == 'Beneficiary To Supporter':
+                # TODO: take into account the language sent
```

Comment:
It was more a TOCO (to correct) than a TODO. If I remember correctly, you said that you will need to wait on the US to know exactly what you need